lib/coney/rabbit_connection.ex

```diff
@@ -3,29 +3,36 @@ defmodule Coney.RabbitConnection do
 
   require Logger
 
-  def open(settings = %{url: url, timeout: timeout}) do
-    case connect(url) do
+  @timeout 1000
+
+  def open(settings) do
+    case connect(settings) do
       {:ok, conn} ->
-        Logger.debug("#{__MODULE__} (#{inspect(self())}) connected to #{url}")
+        Logger.debug("#{__MODULE__} (#{inspect(self())}) connected to server")
 
         Process.monitor(conn.pid)
         conn
 
       {:error, error} ->
         Logger.error(
-          "#{__MODULE__} (#{inspect(self())}) connection to #{url} refused: #{inspect(error)}"
+          "#{__MODULE__} (#{inspect(self())}) connection to server refused: #{inspect(error)}"
         )
 
-        :timer.sleep(timeout)
+        :timer.sleep(@timeout)
```

Comment:
I think someone can use this setting, maybe we can make move this setting from rabbitmq specific settings and make it general?

Reply:
Sure, but be aware of [the warning](https://hexdocs.pm/elixir/Process.html#sleep/1l):

> Use this function with extreme care. For almost all situations where you would use sleep/1 in Elixir, there is likely a more correct, faster and precise way of achieving the same with message passing.

Reply:
Currently it solves the case if connection went down and someone wants to use connection, they will wait for reconnection or return with timeout error if connection wasn't restored in reasonable time. It may look dirty but works, would be really great if you have some ideas how to make it better!